arch/arm/core/cortex_r/Kconfig

```diff
@@ -0,0 +1,98 @@
+# Kconfig - ARM Cortex-R platform configuration options
+
+#
+# Copyright (c) 2018 Marvell
+# Copyright (c) 2018 Lexmark International, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# NOTE: We have the specific core implementations first and outside of the
+# if CPU_CORTEX_R block so that SoCs can select which core they are using
+# without having to select all the options related to that core.  Everything
+# else is captured inside the if CPU_CORTEX_R block so they are not exposed
+# if one selects a different ARM Cortex Family (Cortex-A or Cortex-M)
+
+
+config CPU_CORTEX_R4
+	bool
+	select CPU_CORTEX_R
+	select ARMV7_R
+	select ARMV7_M_ARMV8_M_FP if CPU_HAS_FPU
```

Comment:
ARMV7_M_ARMV8_M_FP  is for Cortex-M CPUs

Reply:
I'm sorry.  Most of this is just copy-paste crud that I'll get rid of.